tests/deploy/test_scheduler_agent.py

```diff
@@ -3137,12 +3138,199 @@ async def restore_baseline_state() -> None:
     assert scheduler._state.resource_state[rid5].deployment_result is DeploymentResult.NEW
     assert scheduler._state.resource_state[rid5].blocked is BlockedStatus.NO
 
-    # TODO: add more scenarios
-    # - resource becomes undefined (in first, second, or last version)
-    # - resource becomes defined (in first, second, or last version)
-    # - resource becomes undefined and then defined again and vice versa
-    # - resource is new+undefined / updated+undefined
-    # - verify requires => like resources, should simply be `== all_models[-1].requires`, same for provides
+    await restore_baseline_state()
+
+    # Resource becomes undefined
+    await scheduler._new_version(
+        [
+            model(
+                {
+                    rid1: all_models[-1].resources[rid1],
+                    rid2: all_models[-1].resources[rid2],
+                    rid4: make_resource_minimal(rid4, values={"new": "unknown"}, requires=[]),
+                },
+                undefined={rid4},
+            ),
+            model(
+                {
+                    rid1: make_resource_minimal(rid1, values={"changed": "unknown"}, requires=[]),
+                    rid2: all_models[-1].resources[rid2],
+                    rid4: make_resource_minimal(rid4, values={"new": "unknown"}, requires=[]),
```

Comment:
You could simply use `all_models[-1].resources[rid4]` here. The purpose of that list is to keep the focus on what changes each version, rather than the full verbose thing.

That said, if you prefer it like this, I won't stand in the way.